The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6149771438/job/16686321302) [(plain)](https://github.com/rust-lang/rust/commit/21e8d6322d06c089ee80e9a22f62a3f4d41bf238/checks/16686321302/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    /cargo/registry/src/index.crates.io-6f17d22bba15001f/ui_test-0.18.1/src/lib.rs:349:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
-  --> $DIR/large_futures.rs:11:9
-   |
-LL |         big_fut([0u8; 1024 * 16]).await;
-   |         ^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider `Box::pin` on it: `Box::pin(big_fut([0u8; 1024 * 16]))`
-   = note: `-D clippy::large-futures` implied by `-D warnings`
-   = note: `-D clippy::large-futures` implied by `-D warnings`
-   = help: to override `-D warnings` add `#[allow(clippy::large_futures)]`
-error: large future with a size of 16386 bytes
-  --> $DIR/large_futures.rs:15:5
-   |
-LL |     f.await
-LL |     f.await
-   |     ^ help: consider `Box::pin` on it: `Box::pin(f)`
-error: large future with a size of 16387 bytes
-  --> $DIR/large_futures.rs:20:9
-   |
-LL |         wait().await;
-LL |         wait().await;
-   |         ^^^^^^ help: consider `Box::pin` on it: `Box::pin(wait())`
-error: large future with a size of 16387 bytes
-  --> $DIR/large_futures.rs:25:13
-   |
-LL |             wait().await;
-LL |             wait().await;
-   |             ^^^^^^ help: consider `Box::pin` on it: `Box::pin(wait())`
 error: large future with a size of 65540 bytes
-  --> $DIR/large_futures.rs:33:5
-   |
-LL |     foo().await;
-LL |     foo().await;
-   |     ^^^^^ help: consider `Box::pin` on it: `Box::pin(foo())`
-error: large future with a size of 49159 bytes
-  --> $DIR/large_futures.rs:35:5
-   |
-   |
-LL |     calls_fut(fut).await;
-   |     ^^^^^^^^^^^^^^ help: consider `Box::pin` on it: `Box::pin(calls_fut(fut))`
-error: large future with a size of 65540 bytes
   --> $DIR/large_futures.rs:48:5
    |
... 6 lines skipped ...
... 6 lines skipped ...
    | |_____^
    |
+   = note: `-D clippy::large-futures` implied by `-D warnings`
+   = help: to override `-D warnings` add `#[allow(clippy::large_futures)]`
 help: consider `Box::pin` on it
... 29 lines skipped ...
    |
error: test failed, to rerun pass `--test compile-test`
 
 
-error: aborting due to 8 previous errors
+error: aborting due to 2 previous errors
 


error: substring `large future with a size of 16385 bytes` not found in stderr output
   |
   |
12 |         //~^ ERROR: large future with a size of 16385 bytes
   |


error: substring ``-D clippy::large-futures` implied by `-D warnings`` not found in stderr output
   |
   |
13 |         //~| NOTE: `-D clippy::large-futures` implied by `-D warnings`
   |


error: substring `large future with a size of 16386 bytes` not found in stderr output
   |
   |
16 |     //~^ ERROR: large future with a size of 16386 bytes
   |


error: substring `large future with a size of 16387 bytes` not found in stderr output
   |
   |
21 |         //~^ ERROR: large future with a size of 16387 bytes
   |


error: substring `large future with a size of 16387 bytes` not found in stderr output
   |
   |
26 |             //~^ ERROR: large future with a size of 16387 bytes
   |


error: substring `large future with a size of 65540 bytes` not found in stderr output
   |
   |
34 |     //~^ ERROR: large future with a size of 65540 bytes
   |


error: substring `large future with a size of 49159 bytes` not found in stderr output
   |
   |
36 |     //~^ ERROR: large future with a size of 49159 bytes
   |

full stderr:
error: large future with a size of 65540 bytes
error: large future with a size of 65540 bytes
##[error]  --> tests/ui/large_futures.rs:48:5
   |
LL | /     async {
LL | |
LL | |         let x = [0i32; 1024 * 16];
LL | |         async {}.await;
LL | |     }
   | |_____^
   |
   = note: `-D clippy::large-futures` implied by `-D warnings`
   = note: `-D clippy::large-futures` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::large_futures)]`
help: consider `Box::pin` on it
   |
LL ~     Box::pin(async {
LL +
LL +         let x = [0i32; 1024 * 16];
LL +         async {}.await;
LL +         println!("{:?}", x);
LL +     })

error: large future with a size of 65540 bytes
##[error]  --> tests/ui/large_futures.rs:60:13
   |
   |
LL | /             async {
LL | |                 let x = [0i32; 1024 * 16];
LL | |                 async {}.await;
LL | |                 println!("macro: {:?}", x);
   | |_____________^
...
...
LL |       macro_!().await
   |
   = note: this error originates in the macro `macro_` (in Nightly builds, run with -Z macro-backtrace for more info)
   = note: this error originates in the macro `macro_` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider `Box::pin` on it
   |
LL ~             Box::pin(async {
LL +                 let x = [0i32; 1024 * 16];
LL +                 async {}.await;
LL +                 println!("macro: {:?}", x);
LL +             })

error: aborting due to 2 previous errors


```

</details>

--- a/compiler/rustc_ty_utils/src/layout.rs
+++ b/compiler/rustc_ty_utils/src/layout.rs
@@ -36,6 +36,9 @@ fn layout_of<'tcx>(
     let (param_env, ty) = query.into_parts();
     debug!(?ty);
 
+    // Optimization: We convert to RevealAll and convert opaque types in the where bounds
+    // to their hidden types. This reduces overall uncached invocations of `layout_of` and
+    // is thus a small performance improvement.
     let param_env = param_env.with_reveal_all_normalized(tcx);
     let unnormalized_ty = ty;
 

